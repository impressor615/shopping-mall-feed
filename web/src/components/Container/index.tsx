import React from "react";

interface ContainerProps {
  children: React.ReactNode;
}

const Container = ({ children }: ContainerProps) => (
  <section className="container">{ children }</section>
);

export default Container;
