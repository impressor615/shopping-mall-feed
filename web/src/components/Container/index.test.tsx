import enzyme from "enzyme";
import React from "react";

import Container from "@/components/Container";

describe("Container", () => {
  const children = <div>children</div>;
  it("should render children", () => {
    const wrapper = enzyme.shallow(<Container>{ children }</Container>);
    expect(wrapper.find(".container")).toHaveLength(1);
    expect(wrapper.contains(children)).toEqual(true);
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<Container>{ children }</Container>);
    expect(wrapper).toMatchSnapshot();
  });
});
