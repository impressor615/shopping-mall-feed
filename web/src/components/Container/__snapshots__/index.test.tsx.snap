// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should be matched with snapshot 1`] = `
<section
  className="container"
>
  <div>
    children
  </div>
</section>
`;
